### PR TITLE
pgadmin: 8.1 -> 8.2, python3Packages.webauthn 1.11.1 -> 2.0.0, python3Packages.flask-security-too: fix pydantic issue

### PR DESCRIPTION
**Pull request by gador**
## Description of changes

This is a major update to pgadmin. (https://www.pgadmin.org/docs/pgadmin4/latest/release_notes_8_2.html)
Mainly, a new `cli` option has been added. 
Due to the change, we can drop most of our `expose-setup` patch.

## Required changes to other packages

- `webauthn` has been updated to version 2. This is a major release (See https://github.com/duo-labs/py_webauthn/blob/master/CHANGELOG.md). No new build failures have been detected, though.
- `flask-security-too`  can drop the `pydantic` broken flag, since `webauthn` has dropped pydantic. This did require an (already merged) patch from upstream

Updating `webauthn` is a more elegant solution IMHO than overwriting `pydantic` version in `pgadmin` and also fixes the until-now broken `flask-security-too`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Module tested successfully in a VM.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by mkg20001 — APPROVED** [review]

**Comment by mkg20001** [comment]
@ofborg test pgadmin

**Comment by mkg20001** [comment]
@ofborg test pgadmin4

**Comment by gador** [comment]
I rebased on current master and found an error, since `flask-security-too` was marked as broken. This was due to a recent merge of https://github.com/NixOS/nixpkgs/pull/271586 (and specifically https://github.com/NixOS/nixpkgs/commit/5fa3852b945e931f21b4e38beb30f3323b7e2e5b). 
I added a fix to use the older version of `pydantic` and it should evaluate now

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 280556` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>nixos-install-tools</li>
</ul>
</details>
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>irrd: log was empty</li>
<li>irrd.dist: log was empty</li>
<li>python312Packages.webauthn: log was empty</li>
<li>python312Packages.webauthn.dist: log was empty</li>
  </ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>pgadmin4</li>
<li>pgadmin4-desktopmode</li>
<li>pgadmin4-desktopmode.dist</li>
<li>pgadmin4.dist</li>
<li>python311Packages.flask-security-too</li>
<li>python311Packages.flask-security-too.dist</li>
<li>python311Packages.webauthn</li>
<li>python311Packages.webauthn.dist</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python311Packages.webauthn.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/mvxadj9c2zyspkn2rl3dhx8js5lghcv0-python3.11-webauthn-1.11.1-dist' from 'daemon'
When evaluating attribute ‘python311Packages.webauthn.dist’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:53:5:

```
   |
53 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python311Packages.webauthn:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/2lvwkf61pb6v7ds42imds2g0js8b2fpd-python3.11-webauthn-1.11.1' from 'daemon'
When evaluating attribute ‘python311Packages.webauthn’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:53:5:

```
   |
53 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.webauthn.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:53:5:

```
   |
53 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.webauthn:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:53:5:

```
   |
53 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
</details>

**Comment by gador** [comment]
Failure in `irrd` is due to a failure in the build dependency `python3Packages.ariadne` which seems unrelated to changes made in this PR:
```
E       AssertionError: assert <snapshottest.pytest.PyTestSnapshotTest object at 0x7fffdf81a110> == [b'Request body cannot be empty']
```
 is a recurring error.

Failure in `python312Packages.webauthn` seems to stem from a build failure of `python3.12-pydantic-1.10.13`. My guess is `pydantic_1` isn't compatible with python 3.12:
```

pydantic>         with pytest.raises(TypeError, match=message):
pydantic> >           Foobar()
pydantic> E           TypeError: Can't instantiate abstract class Foobar without an implementation for abstract method 'get_secret_value'
pydantic>
pydantic> tests/test_types.py:2719: TypeError
pydantic>
pydantic> During handling of the above exception, another exception occurred:
pydantic>
pydantic>     def test_secretfield():
pydantic>         class Foobar(SecretField):
pydantic>             ...
pydantic>
pydantic>         message = "Can't instantiate abstract class Foobar with abstract methods? get_secret_value"
pydantic>
pydantic> >       with pytest.raises(TypeError, match=message):
pydantic> E       AssertionError: Regex pattern did not match.
pydantic> E        Regex: "Can't instantiate abstract class Foobar with abstract methods? get_secret_value"
pydantic> E        Input: "Can't instantiate abstract class Foobar without an implementation for abstract method 'get_secret_value'"
``` 
are the last two of four test failures.

**Comment by gador** [comment]
@ofborg test pgadmin4

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 280556` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>nixos-install-tools</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>irrd: log was empty</li>
<li>irrd.dist: log was empty</li>
  </ul>
</details>
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>pgadmin4</li>
<li>pgadmin4-desktopmode</li>
<li>pgadmin4-desktopmode.dist</li>
<li>pgadmin4.dist</li>
<li>python311Packages.flask-security-too</li>
<li>python311Packages.flask-security-too.dist</li>
<li>python311Packages.webauthn</li>
<li>python311Packages.webauthn.dist</li>
<li>python312Packages.webauthn</li>
<li>python312Packages.webauthn.dist</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python311Packages.webauthn.dist:
<details>

error: build log of '/nix/store/65wn121xx63bk5d00f3v2bp42zjml60m-python3.11-webauthn-2.0.0-dist' is not available
When evaluating attribute ‘python311Packages.webauthn.dist’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:49:5:

```
   |
49 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python311Packages.webauthn:
<details>

error: build log of '/nix/store/mnf64a1fx5qmd1n4s3lz6m1z9byraic1-python3.11-webauthn-2.0.0' is not available
When evaluating attribute ‘python311Packages.webauthn’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:49:5:

```
   |
49 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.webauthn.dist:
<details>

error: build log of '/nix/store/6ccpz5hjcq54ac1w4wvfwmvfmlnyrwzb-python3.12-webauthn-2.0.0-dist' is not available
When evaluating attribute ‘python312Packages.webauthn.dist’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:49:5:

```
   |
49 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
python312Packages.webauthn:
<details>

error: build log of '/nix/store/kyzm0skh9dr1860fj3mnpd18fca3n944-python3.12-webauthn-2.0.0' is not available
When evaluating attribute ‘python312Packages.webauthn’:
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/webauthn/default.nix:49:5:

```
   |
49 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
</details>

**Comment by risicle** [comment]
@ofborg build irrd

Fails for me

**Comment by gador** [comment]
> @ofborg build irrd
> 
> Fails for me

yes, fails on master, too

**Comment by gador** [comment]
@ofborg test pgadmin4

File: nixos/tests/pgadmin4.nix [hunk 1]
```diff
@@ -55,6 +55,12 @@ import ./make-test-python.nix ({ pkgs, lib, ... }:
       machine.wait_until_succeeds("curl -sS localhost:5051/login | grep \"<title>pgAdmin 4</title>\" > /dev/null")
       # check for missing support files (css, js etc). Should catch not-generated files during build. See e.g. https://github.com/NixOS/nixpkgs/pull/229184
       machine.succeed("wget -nv --level=1 --spider --recursive localhost:5051/login")
+      # test idempotenceny
+      machine.systemctl("stop pgadmin.service")
+      machine.systemctl("start pgadmin.service")
```
Comment (SuperSandro2000) [inline nixos/tests/pgadmin4.nix:60]:
```suggestion
      machine.systemctl("restart pgadmin.service")
```

File: pkgs/development/python-modules/flask-security-too/default.nix [hunk 2]
```diff
@@ -59,6 +59,15 @@ buildPythonPackage rec {
     hash = "sha256-we2TquU28qP/ir4eE67J0Nlft/8IL8w7Ny3ypSE5cNk=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "fixes-for-py_webauthn-2.0.patch";
+      url = "https://github.com/Flask-Middleware/flask-security/commit/2abd4e232b7943628f89d5d8033d64896286230f.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-security-too/default.nix:65]:
```suggestion
      url = "https://github.com/Flask-Middleware/flask-security/commit/5725f7021343567ec0b25c890e859f4e84c93ba6.patch";
```

that is the merged hash which has a reference to the PR on https://github.com/Flask-Middleware/flask-security/commit/5725f7021343567ec0b25c890e859f4e84c93ba6

File: pkgs/development/python-modules/flask-security-too/default.nix [hunk 3]
```diff
@@ -59,6 +59,15 @@ buildPythonPackage rec {
     hash = "sha256-we2TquU28qP/ir4eE67J0Nlft/8IL8w7Ny3ypSE5cNk=";
   };
 
+  patches = [
+    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/flask-security-too/default.nix:63]:
```suggestion
    # https://github.com/Flask-Middleware/flask-security/pull/901
    (fetchpatch {
```
just for reference

**Review by SuperSandro2000 — APPROVED** [review]

**Review by mkg20001 — APPROVED** [review]

**Pull request merged**